scipy/stats/_distn_infrastructure.py

```diff
@@ -1540,6 +1541,23 @@ def _penalized_nnlf(self, theta, x):
         n_log_scale = len(x) * log(scale)
         return self._nnlf_and_penalty(x, args) + n_log_scale
 
+    def _get_shape_info(self):
+        return self._shape_info()
+
+
+class ShapeInfo:
+    def __init__(self, shape_name, integrality=False, domain=(-np.inf, np.inf),
+                 inclusive=(True, True)):
+        self.shape_name = shape_name
+        self.integrality = integrality
+
+        domain = list(domain)
+        if np.isfinite(domain[0]) and not inclusive[0]:
+            domain[0] = np.nextafter(domain[0], np.inf)
```

Comment:
Interesting function 👍

Reply:
@andyfaff told me about it. Thanks again @andyfaff!